src/openforms/js/components/formio_builder/translation.js

```diff
@@ -1,3 +1,254 @@
+ * @type {Object} Mapping of type to list of strings or regular expression patterns to
+ *   test the property path against.
+ */
+const ADDITIONAL_PROPERTIES_BY_COMPONENT_TYPE = {
+  textfield: ['defaultValue'],
+  textarea: ['defaultValue'],
+  content: ['html'],
+  fieldset: ['legend'],
+  editgrid: ['groupLabel'],
+  // manually entered data values -> labels are translatable
+  select: [/data\.values\[[0-9]+\]\.label/],
+  selectboxes: [/values\[[0-9]+\]\.label/],
+  radio: [/values\[[0-9]+\]\.label/],
+};
+
+const isRegex = item => {
+  return typeof item.test === 'function';
+};
+
+/**
+ * Determine if the property/path is a component property that is translated.
+ *
+ * This maps to semantics of Formio itself, depending on whether this.t or ctx.t is
+ * called on the _value_ of the property during rendering of the form. We actually
+ * perform these same translations in the backend, but stick to the Formio mechanisms
+ * to stay compatible.
+ *
+ * NOTE: keep this in sync with the backend, see relevant test case:
+ * openforms.submissions.tests.test_submission_report:
+ * DownloadSubmissionReportTests.test_report_is_generated_in_same_language_as_submission
+ *
+ * @param  {String} componentType Type of the formio component. Different types may have different relevant properties.
+ * @param  {String} property      The formio component path, e.g. 'label' or 'validate.required'
+ * @return {Boolean}              True if the value can be translated, False otherwise.
+ */
+export const isTranslatableProperty = (componentType, property) => {
+  // we have a different UI for the content/html translations, relying on only ever
+  // one property being translatable.
+  if (componentType === 'content') {
+    return property === 'html';
+  }
+
+  const allTranslatableProperties = ALWAYS_TRANSLATABLE_PROPERTIES.concat(
+    ADDITIONAL_PROPERTIES_BY_COMPONENT_TYPE[componentType] || []
+  );
+  return allTranslatableProperties.some(entry => {
+    // entry is either a string or a regex. A regex has the 'test' method.
+    if (isRegex(entry)) {
+      return entry.test(property);
+    }
+    // otherwise -> strict string equality check
+    return entry === property;
+  });
+};
+
+/**
+ * Update the container holding all the component translations with the translations
+ * of a particular component.
+ *
+ * @param  {Object} container Key-value mapping, key is language code, values are mappings
+ *                            of literals to their translations *for that language code*.
+ * @param  {Object} component Formio component/schema - it contains the configuration
+ *                            of one particular component in the form builder.
+ * @return {Void}             The container is mutated directly. It is advised to pass
+ *                            in an Immer draft or carefully manage mutable objects.
+ */
+export const persistComponentTranslations = (container, component) => {
```

Comment:
Reading `persistComponentTranslations` elsewhere without context, made me think persistence to the backend. While it's actually moving data from the passed in component to the container... Can't really think of a better name :thinking: